src/day2.scala

```diff
@@ -0,0 +1,63 @@
+// using scala 3.0.2
+
+package day2
+
+import scala.io.Source
+
+@main def part1(): Unit =
+  val input = Source.fromFile("input/day2").mkString
+  val answer = part1(input)
+  println(s"The solution is $answer")
+
+@main def part2(): Unit =
+  val input = Source.fromFile("input/day1").mkString
+  val answer = part2(input)
+  println(s"The solution is $answer")
+
+def part1(input: String): Int =
+  val entries = input.linesIterator.flatMap(Direction.from)
+  val firstPosition = Position(0, 0)
+  val lastPosition = entries.foldLeft(firstPosition)((position, direction) =>
+    position + direction
+  )
+  lastPosition.result
+
+def part2(input: String): Int =
+  val entries = input.linesIterator.flatMap(Direction.from)
```

Comment:
What does the specification says? Are you supposed to silently ignore parsing failures, or stop the program?

I personally think it is bad practice to swallow parsing failures like so. I would rather make them stop the program early, instead of producing an empty list of `entries`.

Reply:
I don't like either.
there is no method in the standard library to do Seq[Option] to Option[Seq]
I m planning to change that ^^

Reply:
In this situation, I would suggest failing early, instead of dealing with `Option`.